indico/modules/events/registration/google_wallet.py

```diff
@@ -0,0 +1,272 @@
+
+from indico.core import signals
+from indico.core.config import config
+from indico.core.logger import Logger
+
+
+class GoogleWalletManager:
+    """Google Wallet management class.
+
+    """
+
+    def __init__(self, event, registration=None):
+        self.event = event
+        self.registration = registration
+        self.settings = self.get_google_wallet_settings(self.event.category)
+        self.credentials = None
+        self.http_client = None
+        self.base_url = 'https://walletobjects.googleapis.com/walletobjects/v1'
+        self.batch_url = 'https://walletobjects.googleapis.com/batch'
+        self.class_url = f'{self.base_url}/eventTicketClass'
+        self.object_url = f'{self.base_url}/eventTicketObject'
+
+        # Set up authenticated client
+        if self.settings:
+            self.settings['google_wallet_application_credentials'] = json.loads(
+                self.settings['google_wallet_application_credentials'])
+            self.auth()
+
+    @staticmethod
+    def get_google_wallet_settings(category):
+        while category:
+            if category.google_wallet_settings:
+                return category.google_wallet_settings
+            else:
+                category = category.parent
+        return {}
+
+    def auth(self):
+        """Create authenticated HTTP client using a service account file."""
+        self.credentials = Credentials.from_service_account_info(
+            self.settings['google_wallet_application_credentials'],
+            scopes=['https://www.googleapis.com/auth/wallet_object.issuer'])
+        self.http_client = AuthorizedSession(self.credentials)
+
+    def create_class_template(self, issuer_id: str, class_suffix: str) -> dict:
+        """ This method will return a dict format ticket class template."""
+        dict_template = {
+            'id': f'{issuer_id}.{class_suffix}',
+            'issuerName': self.settings['google_wallet_issuer_name'],
+            'reviewStatus': 'UNDER_REVIEW',
+            'eventName': {
+                'defaultValue': {
+                    'language': 'en-US',
+                    'value': self.event.title
+                }
+            },
+            'venue': {
+                'name': {
+                    'defaultValue': {
+                        'language': 'en-US',
+                        'value': self.event.venue_name if not self.event.room_name
+                        else f'{self.event.venue_name} - Room {self.event.room_name}'
+                    }
+                },
+                'address': {
+                    'defaultValue': {
+                        'language': 'en-US',
+                        'value': self.event.address
+                    }
+                }
+            },
+            'dateTime': {
+                'start': self.event.start_dt_local.isoformat(),
+                'end': self.event.end_dt_local.isoformat()
+            },
+            'customGateLabel': {
```

Comment:
I thought we had agreed to try to avoid using "gate" and "seat" fields for custom content? Is there no better option? From a semantic point of view this is not great because you don't know if Google at some points starts showing e.g. icons related to the purpose of those fields...

Reply:
So, what I can do is to OVERRIDE the standard template (https://developers.google.com/wallet/tickets/events/resources/template?hl=en) in the 2nd and 3rd row, so that we are sure that Even if they change something, our part will not be affected.
I also took the opportunity to add the END DATE of the event (replacing the starting time... let me know if you prefer to display START/END date in another format).

Reply:
What are the disadvantages of overriding?

And I think the start date (or both dates) is much more useful than showing (just) the end date. When does the end date actually matter? You're already at the venue, usually you'll know when the event finishes anyway...

Reply:
Disadvantages: you have to deal yourself with label translations... I think no much more than that.
Start/End date: we (our Security) wants to have Start and End date displayed... but that's not an issue: we will customize the Class template... so I can keep for the "Cern" version whatever you prefer :)
- Start Date - Start Time
or
- Start Date - End Date
or 
- Start DateTime - End DateTime
Please, choose :)

Reply:
Would the logic we use on meeting pages work for you guys? It's kind of smart depending on whether it's single-day or multipday. Check `indico/modules/events/templates/display/indico/_common.html` for the logic.

Reply:
Ok, I'm not sure I have been able to correctly translate from the date formatting... but at least this should work for us.
Please, feel free to change it accordingly to your needs... I'll customize it later in our UN plugin hooking the signal.

Reply:
There's actually also `format_event_date` in `indico/modules/events/templates/display/conference/_util.html` which gives (imo) a nicer output e.g.:

currently: `19 Sept 2023 - 21 Sept 2023`
with `format_event_date`: `19-21 Sept 2023`

Reply:
Ok, updated.